R/derive_vars_dtm_to_tm.R

```diff
@@ -0,0 +1,64 @@
+#' Derive time variables from datetime variables
+#'
+#' This function creates a time as output from a datetime variable
+#'
+#' @param dataset Input dataset
+#'
+#' @param source_vars A list of datetime variables from which time are to be extracted
+#'
+#' @param hms_flag A `"Y"` scalar character. If `"Y"`, time is represented as `"HH:MM:SS"`
+#'  otherwise they are presented as `"HH:MM"`.
+#'  Optional; default is `NULL`.
+#'
+#' @author Teckla Akinyi
+#'
+#' @return A data frame containing all observations and variables of the input
+#'   dataset and adds the corresponding time variable of all specified datetime variables from
+#'   the source_vars option with the correct name.
+#'
+#' @keywords ADaM Timing Date
+#'
+#' @export
+#'
+#' @examples
+#' library(dplyr, warn.conflicts = FALSE)
+#' library(lubridate, warn.conflicts = FALSE)
+#'
+#' adcm <- tibble::tribble(
+#'  ~STUDYID, ~USUBJID, ~TRTSDTM,               ~ASTDTM,               ~AENDTM,
+#'  "TEST01", "PAT01",  "2012-02-25 23:41:10", "2012-02-28 19:03:00", "2013-02-25 23:32:16",
+#'  "TEST01", "PAT01",  "",                    "2012-02-28 19:00:00", "",
+#'  "TEST01", "PAT01",  "2017-02-25 23:00:02", "2013-02-25 19:00:15", "2014-02-25 19:00:56",
+#'  "TEST01", "PAT01",  "2017-02-25 16:00:00", "2017-02-25 14:25:00", "2017-03-25 23:00:00",
+#'  "TEST01", "PAT01",  "2017-02-25 16:05:17", "2017-02-25 14:20:00", "2018-04-29 14:06:45",
+#'  ) %>% mutate(
+#'  TRTSDTM = lubridate::as_datetime(TRTSDTM),
+#'  ASTDTM = lubridate::as_datetime(ASTDTM),
+#'  AENDTM = lubridate::as_datetime(AENDTM)
```

Comment:
No need to use the `lubridate::` prefix here as you've already loaded the package above.